### DISCUSSION TITLE
Starting point for the use of maxGraph Editor

### DISCUSSION THREAD

**Discussion by io226041** [maxGraph/maxGraph#99] [Q&A]
Hello,

I intend to use the maxGraph editor as a basis for my project (the goal: maxGraph editor should serve as a configuration tool for the UI of a Java app). As a start, I would like to use the editor and then adapt it for the purposes of my project. A suitable example for this would be [this grapheditor](https://github.com/jgraph/mxgraph/blob/master/javascript/examples/grapheditor/www/index.html).
In the examples of maxGraph I unfortunately don't see an example with a complete editor with toolbar and sidebar, but only how editor is used without toolbar and sidebar, e.g. here: [SwimLanes.stories.js](https://github.com/maxGraph/maxGraph/blob/development/packages/html/stories/SwimLanes.stories.js#L45).

One more thing is unclear to me. Is the use of the xml configuration files as described at [EditorCodec.html](https://github.com/maxGraph/maxGraph/blob/development/docs/api/classes/EditorCodec.html) or also described at [tutorial.html](https://github.com/maxGraph/maxGraph/blob/development/docs/old/tutorial.html#L324) current or deprecated? Both in the examples of [mxGraph](https://github.com/jgraph/mxgraph/blob/master/javascript/examples/grapheditor/www/index.html#L102) and [drawio](https://github.com/jgraph/drawio/blob/dev/src/main/webapp/js/diagramly/App.js#L920) the approach of xml configuration is not followed, but Editor instances are created using an alternative concept without xml. In the examples, maxGraph also uses the editor constructor without xml-config ([Swimlanes.stories.js](https://github.com/maxGraph/maxGraph/blob/development/packages/html/stories/SwimLanes.stories.js#L45)), although the xml-config approach is commented out directly above it.

I find the approach of configuring the sidebar, toolbar and context menu via xml files very interesting, but unfortunately have not found any working examples anywhere.

Could someone give me a hint on how best to put together a working editor as a starting point? It is important for me to know whether configuring the editor, toolbar and sidebar via xml makes sense or whether the xml configuration method is deprecated and should no longer be used?

**Comment by tbouffard** [comment]
Hi AFAIK, the graph editors examples haven't been migrated to the new maxGraph API. There are still in "packages/html/stories/stashed".
There is currently no decision about if these editors are deprecated or will be supported. 
The same applies to xml configuration (which involves using "eval").

So, if you are interested in such support, I suggest you to play with it and try to make it works.

**Reply by sadek-hussein-yuvo** [reply]
@tbouffard any update about graph editors examples ?

**Reply by tbouffard** [reply]
hello @sadek-hussein-yuvo 
I don't think so: https://github.com/maxGraph/maxGraph/tree/development/packages/html/stories/stashed

**Reply by mayorovad** [reply]
Hi @sadek-hussein-yuvo 
Most of editor examples starts with `.xml` config loading and parsing, which will not work until #125 will be looked into.
Another issue is that now `Graph` instance is strictly attached to DOM container (in `mxgraph` it's not mandatory). Even if you not pass DOM element to constructor of `Graph`, it's still create some `div` and put graph in it. Editor examples are based on creating editor with `Graph` instance without DOM, then attaching it with method `setContainer` which called `Graph.init` that is lost during poring I suppose.
https://github.com/maxGraph/maxGraph/blob/5b1369b0193f24be80bcfc4ec9c7815fe87c10df/packages/core/src/editor/Editor.ts#L1536-L1542

The simplest solution I can imagine is to add DOM container to `Editor` constructor arguments and see what will be 🤔

**Reply by sadek-hussein-yuvo** [reply]
@mayorovad  so for now we need to hold the use of mxgraph  till we have a stable version !, maybe it's better to use the js one (maxgraph).
it's not easy to add a DOM container to the Editor contractor maybe it will affect other features.
in all cases, thanks got your help.

**Reply by tbouffard** [reply]
@mayorovad thanks for noting that the TypeScript migration removed the ability to attach the Graph to the DOM container afterwards.
I don't know how/when/if this is useful but I am not sure it has been done on purpose, so we could at least create an issue to track the change.

**Reply by tbouffard** [reply]
See #367

**Comment by janhlavsa** [comment]
Hi guys, is there please any movement with this?

**Reply by tbouffard** [reply]
Hi, the GraphEditor examples has not been reworked.
There are some examples using the Editor features. For example, see https://github.com/maxGraph/maxGraph/blob/v0.7.0/packages/html/stories/Toolbar.stories.ts
